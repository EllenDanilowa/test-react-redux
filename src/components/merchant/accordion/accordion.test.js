import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Accordion from './accordion';
import {
  Label,
  ArrowUp
} from './accordion.styled';

describe('Accordion', () => {
  let element;
  let title;
  let childrenId;
  let children;

  const createElement = ({title, children}, isSnapshot = true) => {
    const template = (<Accordion title={title}>{children}</Accordion>);

    return isSnapshot ?
      renderer.create(template).toJSON() :
      shallow(template);
  };

  beforeEach(() => {
    title = 'Title';
    childrenId = '#children';
    children = React.createElement('p', {id: childrenId.slice(1)}, 'accordion content');
  });

  it('renders component correctly', () => {
    element = createElement({title, children});

    expect(element).toMatchSnapshot();
  });

  it('shows content if opened', () => {
    element = createElement({title, children}, false);

    element.find(Label).simulate('click');

    expect(element.find(ArrowUp).length).toBe(1);
    expect(element.find(childrenId).length).toBe(1);
  });
});
