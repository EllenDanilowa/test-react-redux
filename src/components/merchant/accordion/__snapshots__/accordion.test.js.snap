// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders component correctly 1`] = `
<div>
  <button
    className="sc-AxhCb kOrOqp"
    onClick={[Function]}
  >
    Title
    <i
      className="sc-AxjAm sc-AxirZ qnlCY"
    />
  </button>
</div>
`;
